### PR TITLE
Sonos binding: add SHUFFLE and REPEAT controls

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
This PR implements one part of the request #1617.

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml [hunk 1]
```diff
@@ -252,4 +252,18 @@
 		<state min="0" max="68399" step="1" readOnly="false">
 		</state>
 	</channel-type>
+
+	<channel-type id="shuffle">
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml]: I'd suggest to mark this "advanced".

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml [hunk 2]
```diff
@@ -252,4 +252,18 @@
...
+		<item-type>Switch</item-type>
+		<label>Shuffle</label>
+		<description>Shuffle queue playback</description>
+		<category>Switch</category>
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml]: remove the category

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml [hunk 3]
```diff
@@ -252,4 +252,18 @@
...
+	</channel-type>
+
+	<channel-type id="repeat">
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml]: I'd suggest to mark this "advanced".

**Comment by kaikreuzer** [comment]
@kgoderis & @R-Edelhoff Could you review and test?

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml (hunk 2 above)
Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml]: All channels of type Switch in this binding have the switch category set. Should we suppress the category more generally for the channels of this binding ?

Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/thing/channels.xml]: I would say so - I am anyhow not sure what the "Switch" category is exactly meant to express, but having the channel to be linked to a Switch item type should be good enough as the "switch" icon is then anyhow the default that gets used.

**Comment by lolodomo** [comment]
All @kaikreuzer 's comments taken into account.

**Comment by lolodomo** [comment]
@kgoderis: can you review my change please ?

**Comment by kgoderis** [comment]
I did not test the code itself but reading it, it LGTM

**Comment by lolodomo** [comment]
I tezted all combinations myself ;)

**Comment by lolodomo** [comment]
Can this PR be merged ?
I am waiting for it to implement enhancement 1264.

**Comment by maggu2810** [comment]
> @kgoderis & @R-Edelhoff Could you review and test?

@kgoderis had a look at the code and added "LGTM"

Still waiting for a response of @R-Edelhoff.

**Comment by lolodomo** [comment]
2 reviewers required for such a basic change ?!

**Comment by maggu2810** [comment]
I don't know the Sonos code at all, so I cannot review it.

> 2 reviewers required for such a basic change ?!

I don't think so, but as @kaikreuzer asked for the comment of both persons I leave it up to him.

**Comment by R-Edelhoff** [comment]
If the functionallity is fully working as you have already mentioned then it looks fine for me as well

**Comment by maggu2810** [comment]
Thanks :wink:

**Pull request merged**